controllers/cnbi/controller_test.go

```diff
@@ -0,0 +1,155 @@
+						RuntimeEnvironment: meteorv1alpha1.CustomNBImageRuntimeSpec{},
+						PackageVersions:    []string{},
+						BuildTypeSpec:      build,
+					},
+					Status: meteorv1alpha1.CustomNotebookImageStatus{},
+				}
+				Expect(k8sClient.Create(context.Background(), cnbi)).Should(Succeed())
+
+				By("checking the CustomNBImage object has been created on the cluster")
+				// lets give the cluster a little time to start reconciling
+				time.Sleep(20 * time.Second)
+
+				lookupKey := types.NamespacedName{Name: "test-2", Namespace: "default"}
+				createdCNBi := &meteorv1alpha1.CustomNBImage{}
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, lookupKey, createdCNBi)
+					return err == nil
+				}, timeout, interval).Should(BeTrue())
+
+				By("looking if the Controller started reconciling the CustomNBImage object")
+				Expect(createdCNBi.Status.Phase).Should(Equal(meteorv1alpha1.CNBiPhaseRunning))
+				Expect(createdCNBi.Status.Conditions).ShouldNot(BeEmpty())
+				Expect(createdCNBi.Status.Conditions[0].Type).Should(Equal(meteorv1alpha1.ImportingImage))
+			})
+			It("should have Condition 'RequiredSecretMissing' if the import is from a repo that reqs auth and Secret is not ready", func() {
+				By("creating a CustomNBImage object for an import from a private repository")
+				cnbi_name := "import-private-repository"
+
+				importFromPrivate := meteorv1alpha1.BuildTypeSpec{
+					BuildType: meteorv1alpha1.ImportImage,
+					FromImage: "quay.io/thoth-station/s2i-custom-notebook:latest",
+					ImagePullSecret: meteorv1alpha1.ImagePullSecret{
+						Name: "private-repository-credentials",
+					},
+				}
+				cnbi := &meteorv1alpha1.CustomNBImage{
+					TypeMeta:   metav1.TypeMeta{APIVersion: "meteor.zone/v1alpha1", Kind: "cnbi"},
+					ObjectMeta: metav1.ObjectMeta{Name: cnbi_name, Namespace: "default"},
+					Spec: meteorv1alpha1.CustomNBImageSpec{
+						RuntimeEnvironment: meteorv1alpha1.CustomNBImageRuntimeSpec{},
+						BuildTypeSpec:      importFromPrivate,
+					},
+					Status: meteorv1alpha1.CustomNotebookImageStatus{},
+				}
+				Expect(k8sClient.Create(context.Background(), cnbi)).Should(Succeed())
+
+				By("checking the CustomNBImage object has been created on the cluster")
+				// lets give the cluster a little time to start reconciling
+				time.Sleep(8 * time.Second)
+
+				lookupKey := types.NamespacedName{Name: cnbi_name, Namespace: "default"}
+				createdCNBi := &meteorv1alpha1.CustomNBImage{}
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, lookupKey, createdCNBi)
+					return err == nil
+				}, timeout, interval).Should(BeTrue())
+
+				By("looking if the Controller started reconciling the CustomNBImage object")
+				Expect(createdCNBi.Status.Phase).Should(Equal(meteorv1alpha1.CNBiPhaseRunning))
+				Expect(createdCNBi.Status.Conditions).ShouldNot(BeEmpty())
+				Expect(createdCNBi.Status.Conditions[0].Type).Should(Equal(meteorv1alpha1.RequiredSecretMissing))
+			})
+		}) */
```

Comment:
If it's commented out, maybe it should not be in this PR but in a future one ? Not fan of including commented code.